File: 2023 yard hoof cup/C++ The end scene/1.cpp
#include <bits/stdc++.h>

using namespace std;
#define int long long

int n, t;
vector<int> arr;

void solve() {
    cin >> n >> t;
    arr.assign(n, 0);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    int day = 0;
    for (int i = 0; i < n && t > 0; i++, day++) {
        int diff = 1440 - arr[i];
        if (diff >= 60) {
            t -= diff;
        }
    }
    cout << day << endl;
}

auto main() -> signed {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);

    solve();
    return 0;
}

File: 2023 yard hoof cup/Python The end scene/1.py
def solve():
    n, t = map(int, input().split())
    arr: list[int] = list(map(int, input().split()))
    day = 1
    for it in arr:
        diff = 1440 - it
        if diff >= 60:
            t -= diff
        if t <= 0:
            break
        day += 1

    print(day)


if __name__ == '__main__':
    solve()
